### PR TITLE
readds nullcrates, this time supplied by clown planet :o)

### PR DESCRIPTION
**Pull request by plapatin**
consider this my redemption arc

## About The Pull Request
nullcrates are back, but this time they give you clownop gear instead of traitor gear

## Why It's Good For The Game
people kept complaining
clownop gear is criminally underused, only being available in an admin-only gamemode
helps distinguish between whos a traitor and whos a dumbass with cargo access
something something improve not remove

## Changelog
:cl:
add: nullcrates are back, this time they have clownop gear in them
/:cl:

### CODE REVIEW

**Comment by plapatin** [comment]
jesus christ every single one of my edits is a merge conflict what the fuck

**Comment by Memesaredreams** [comment]
Plap is orange's puppet wake up sheeple

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -155,7 +155,7 @@
 		var/category = pick(uplink_items)
 		var/item = pick(uplink_items[category])
 		var/datum/uplink_item/I = uplink_items[category][item]
-		if(!I.surplus_nullcrates || prob(100 - I.surplus_nullcrates))
+		if ( !(/datum/game_mode/nuclear/clown_ops in I.include_modes) || !I.surplus_nullcrates || prob(100 - I.surplus_nullcrates))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/cargo/packs.dm]:
```
/datum/supply_pack/emergency/syndicate/fill(obj/structure/closet/crate/C)
    var/crate_value = 30
    var/list/uplink_items = GLOB.uplink_items
    while(crate_value)
        var/category = pick(uplink_items)
        var/item = pick(uplink_items[category])
        var/datum/uplink_item/I = uplink_items[category][item]
        if (!I.surplus_nullcrates || prob(100 - I.surplus_nullcrates))
```

**Comment by ExcessiveUseOfCobblestone** [comment]
revert all your surplus changes, those have unintentional side effects.

add `var/surplus_nullcrates` again

do NOT add
```DM
/datum/uplink_item/New()
    . = ..()
    if(isnull(surplus_nullcrates)
        surplus_nullcrates = surplus
```

then just edit the things you want in the null crate by editing surplus_null to a number. Honestly though the whole loop is kinda intuitive and I'd rather it just be a pickweight a la

```DM
/datum/supply_pack/emergency/syndicate/fill(obj/structure/closet/crate/C)
    var/static/list/nullcrate_picklist
    if(isnull(nullcrate_picklist)
        for(var/i in GLOB.uplink_items)
            var/datum/uplink_item/the_uplink_item = i
            if(!the_uplink_item.surplus_nullcrates)
                continue
            nullcrate_picklist += list(the_uplink_item = the_uplink_item.surplus_nullcrates)
    var/crate_value = 30
    while(crate_value)
        var/datum/uplink_item/chosen_uplink_item = pickweight(nullcrate_picklist)
        if(chosen_uplink_item.cost > crate_value)
            continue
        crate_value -= chosen_uplink_item.cost
        new chosen_uplink_item.item(C)
```

That means if you have object A have `surplus_nullcrates` of 10 while B has a `surplus_nullcrates` of 20, B will be twice as likely to spawn as A does if the crate_value is high enough. You could also set it up so that the item loses likelihood to spawn the more it spawns and stuff like that.

**Comment by plapatin** [comment]
fuck

**Comment by adamkad1** [comment]
plz fix this, is a good idea

**Comment by wesoda25** [comment]
nooo plappers dont delete!!

**Pull request closed without merging**